src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PropertyAccess.cs

```diff
@@ -94,13 +95,23 @@ private BoundExpression MakePropertyGetAccess(SyntaxNode syntax, BoundExpression
                 Debug.Assert(getMethod.ParameterCount == rewrittenArguments.Length);
                 Debug.Assert(getMethodOpt is null || ReferenceEquals(getMethod, getMethodOpt));
 
-                return BoundCall.Synthesized(
+                ArrayBuilder<LocalSymbol>? temps = null;
+                rewrittenReceiver = AdjustReceiverForExtensionsIfNeeded(rewrittenReceiver, getMethod, storesOpt: null, ref temps);
```

Comment:
>rewrittenReceiver = AdjustReceiverForExtensionsIfNeeded(rewrittenReceiver, getMethod, storesOpt: null, ref temps); [](http://example.com/codeflow?start=16&length=114)

What is the scenario for which we need to to do this?

Reply:
A property get access. See `AdjustReceiver_PropertyAccess`

Reply:
I do not think this logic belongs in this helper. By the time we get here the receiver is rewritten and should have all appropriate transforms applied. For example, for a regular property get access the transform should be handled by VisitPropertyAccess`, I think.